components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/CaptureBuildScansListener.java

```diff
@@ -76,8 +78,8 @@ private void capturePublishedBuildScan(BuildScanApi buildScan, MavenProject root
             String baseUrl = String.format("%s://%s%s", scan.getBuildScanUri().getScheme(), scan.getBuildScanUri().getHost(), port);
 
             try (FileWriter fw = new FileWriter(EXPERIMENT_DIR + "/build-scans.csv", true);
-                 BufferedWriter bw = new BufferedWriter(fw);
-                 PrintWriter out = new PrintWriter(bw)) {
+                BufferedWriter bw = new BufferedWriter(fw);
+                PrintWriter out = new PrintWriter(bw)) {
```

Comment:
not related to this PR: my understanding is that we need to close this `Printwriter` which is the outermost writer. Closing the outermost writer will take care of closing the inside ones - in this case `bw` and `fw`.

Reply:
Your comment made me realize that the previous indentation was there on purpose. I'll revert my change.

The current implementation is fine as the writers are part of the `try-with-resources` definition and will all be closed in reverse order of definition.
Closing an already closed stream is a `no-op`.

Reply:
ah i see that now, i missed that they were all inside the try-with-resources :D